assets/js/blocks/collection-filters/inner-blocks/stock-filter/frontend.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import {
+	store as interactivityStore,
+	navigate,
+} from '@woocommerce/interactivity';
+import { DropdownContext } from '@woocommerce/interactivity-components/dropdown';
+import { HTMLElementEvent } from '@woocommerce/types';
+
+const getUrl = ( activeFilters: string ) => {
+	const url = new URL( window.location.href );
+	const { searchParams } = url;
+
+	if ( activeFilters !== '' ) {
+		searchParams.set( 'filter_stock_status', activeFilters );
+	} else {
+		searchParams.delete( 'filter_stock_status' );
+	}
+
+	return url.href;
+};
+
+type StockFilterState = {
+	filters: {
+		stockStatus: string;
+		activeFilters: string;
+		showDropdown: boolean;
+	};
+};
+
+type ActionProps = {
+	state: StockFilterState;
+	event: HTMLElementEvent< HTMLInputElement >;
+};
+
+interactivityStore( {
+	state: {
+		filters: {
+			stockStatus: '',
+		},
+	},
+	actions: {
+		filters: {
+			navigate: ( { context }: { context: DropdownContext } ) => {
+				if ( context.woocommerceDropdown.selectedItem.value ) {
+					navigate(
+						getUrl( context.woocommerceDropdown.selectedItem.value )
+					);
+				}
+			},
+			updateProducts: ( { event }: ActionProps ) => {
```

Comment:
This overrides the method set for price filter. We will need a better way to update the products to handle multiple filters, but for now I think renaming this to an unique name would work for the MVP.

About the better way to do this, I think we should define a context for Collection filter, and a state to hold active filters, an action watch for state change and update the products. Inner filter blocks should mutate the active filters state only.